### PR TITLE
report(performance): use metric savings for metric filter

### PR DESCRIPTION
**Pull request by adamraine**
https://github.com/GoogleChrome/lighthouse/pull/15445#discussion_r1340767373

Metric filter was still using `relevantAudits` which is just a manually curated list that we haven't touched in a while. This PR uses `metricSavings` instead to determine filter applicability and sorts the audits on the filtered metric.

### CODE REVIEW

File: report/renderer/performance-category-renderer.js [hunk 1]
```diff
@@ -338,7 +372,7 @@ export class PerformanceCategoryRenderer extends CategoryRenderer {
 
       const labelEl = this.dom.createChildOf(metricFilterEl, 'label', 'lh-metricfilter__label');
       labelEl.htmlFor = elemId;
-      labelEl.title = metric.result?.title;
+      labelEl.title = 'result' in metric ? metric.result.title : '';
```
Comment (adamraine) [inline report/renderer/performance-category-renderer.js:375]: This fixes a bug where the `title` of the "All" filter button is `"undefined"`

File: types/lhr/lhr.d.ts [hunk 2]
```diff
@@ -197,6 +197,8 @@ declare module Result {
 
   /** Gather mode used to collect artifacts. */
   type GatherMode = 'navigation'|'timespan'|'snapshot';
+
+  type MetricAcronym = 'FCP' | 'LCP' | 'TBT' | 'CLS' | 'INP' | 'SI' | 'TTI' | 'FMP';
```
Comment (adamraine) [inline types/lhr/lhr.d.ts:201]: [Over generalizing for fun](https://www.typescriptlang.org/play?#code/C4TwDgpgBAMhzAgJygXigcgIIYD4YCE8MBhYgEWIFFiAxYgcWIAliBJYgKWIGliZiAWWIA5YgHliABWIBFYgCViAZWIAVYgFViANWIB1YgA1iATWIAtDAG4AsAChQkKFgDGSAPYA7EAFs0UAAGACQA3nAIyAC+YRGISDHh8PFRgXb2Dq7eAM7AUACG7t5+AFwuRT7+6KQwqtZAA)

**Comment by paulirish** [comment]
ping @connorjclark

File: report/renderer/performance-category-renderer.js [hunk 3]
```diff
@@ -211,73 +212,104 @@ export class PerformanceCategoryRenderer extends CategoryRenderer {
...
+        if (
+          a.overallImpact === 0 && b.overallImpact === 0 &&
+          a.overallLinearImpact !== b.overallLinearImpact
+        ) {
+          return b.overallLinearImpact * b.guidanceLevel - a.overallLinearImpact * a.guidanceLevel;
+        }
+
+        // Audits that have no estimated savings should be prioritized by the guidance level
+        return b.guidanceLevel - a.guidanceLevel;
+      }).forEach(item => {
```
Comment (adrianaixba) [inline report/renderer/performance-category-renderer.js:281]: small nit: could this `.foreach` be in a separate line for readability? unless this is generally preferred then it's ok :)

**Review by adrianaixba — APPROVED** [review]
lgtm!

File: report/renderer/category-renderer.js [hunk 4]
```diff
@@ -38,11 +38,11 @@ export class CategoryRenderer {
 
   /**
    * @param {LH.ReportResult.AuditRef} audit
-   * @return {Element}
+   * @return {HTMLElement}
    */
   renderAudit(audit) {
     const component = this.dom.createComponent('audit');
-    return this.populateAuditValues(audit, component);
+    return /** @type {HTMLElement} */ (this.populateAuditValues(audit, component));
```
[the remark this answers is not in the document]
Comment (adamraine) [reply report/renderer/category-renderer.js:45]: https://github.com/GoogleChrome/lighthouse/blob/new-perf-audit-filter/report/renderer/performance-category-renderer.js#L241

`hidden` is not a property on `Element` but it is on `HTMLElement`

**Pull request merged**